pypesto/store/read_from_hdf5.py

```diff
@@ -120,3 +150,94 @@ def read(self) -> Result:
                 self.results.optimize_result.append(result)
             self.results.optimize_result.sort()
         return self.results
+
+
+class SamplingResultHDF5Reader:
+    """
+    Reader of the HDF5 result files written
+    by class SamplingResultHDF5Writer.
+
+    Attributes
+    -------------
+    storage_filename:
+        HDF5 result file name
+    """
+    def __init__(self, storage_filename: str):
+        """
+        Parameters
+        ----------
+
+        storage_filename: str
+            HDF5 result file name
+        """
+        self.storage_filename = storage_filename
+        self.results = Result()
+
+    def read(self) -> Result:
+        """
+        Read HDF5 result file and return pyPESTO result object.
+        """
+        sampleResult = {}
+        with h5py.File(self.storage_filename, "r") as f:
+            if '/problem' in f['/']:
+                problem_reader = ProblemHDF5Reader(self.storage_filename)
+                self.results.problem = problem_reader.read()
+            for key in f['/sampling/results']:
+                sampleResult[key] = \
+                    f[f'/sampling/results/{key}'][:]
+            for key in f['/sampling/results'].attrs:
+                sampleResult[key] = \
+                    f['/sampling/results'].attrs[key]
+        try:
+            self.results.sample_result = McmcPtResult(**sampleResult)
+        except ValueError:
+            # TODO : Warning here
```

Comment:
TODO

Reply:
added 
```python
logger.warning(
                "Could not create a McmcPtResult from sample_result."
                "Will continue with a list.")
```
ok like that?

Reply:
Moderately informative. Why would that happen? Append the original exception?

Reply:
Well, and it's not a list, but a dict. Please also update the return type typehint accordingly.

Not too sure how helpful it is to return the dict instead here. Either all callers will have to handle both types, or will probably fail. Maybe it's easier to just fail here?

Reply:
you might be right that it is easier to fail here. Although since McmcPtResult inherits from dict, it should actually still work fine i think.

Reply:
>  But there is no real reason why self.results.sample_result = McmcPtResult(**sampleResult)should fail so i think i might as well remove the try... except?

Agreed

Reply:
Done :+1: